### PR TITLE
Key results for detail panel

### LINKED ISSUE #52: Calculate key results for detail settings
Calculate key numbers in detail panel, they should change on slider change:

image

Calculation is quite similar to main slider calculation.
You'll need the new `technology_data.json` from the Wolke.

Wind
1. area in ha
2. area * power_density / nominal_power_per_unit (`technology_data.json` -> `nominal_power_per_unit` -> `wind`)
3. area * power_density * full_load_hours (2045)

PV ground
1. area in ha
2. area * power_density * full_load_hours (2045)
with power_density and full_load_hours per technology:
_soil_quality_low_: `power_density` -> `pv_ground`, `full_load_hours` -> `pv_ground`
_soil_quality_medium_: `power_density` -> `pv_ground_vertical_bifacial`, `full_load_hours` -> `pv_ground_vertical_bifacial`
_permanent crops_: `power_density` -> `pv_ground_elevated`, `full_load_hours` -> `pv_ground_elevated`

PV roof
1. area in ha
2. area * power_density * full_load_hours (2045)

(All values rounded to int)

### PR DESCRIPTION
**Pull request by henhuy**
Closes #52 

Waits for ionrangeslider fix or shall I do it using AJAX?

### CODE REVIEW

**Review by nesnoj — COMMENTED** [review]
Thx for this feature!
If you think it's worth to wait for the ion fix, we can give it a couple of more days. But if this solution is ok for you and the other wouldn't be much shorter, go for it.
BTW, currently the app doesn't start at my end:
```
Traceback (most recent call last):
  File "/app/manage.py", line 32, in <module>
    execute_from_command_line(sys.argv)
  File "/venv/lib/python3.9/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/venv/lib/python3.9/site-packages/django/core/management/__init__.py", line 416, in execute
    django.setup()
  File "/venv/lib/python3.9/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/venv/lib/python3.9/site-packages/django/apps/registry.py", line 124, in populate
    app_config.ready()
  File "/app/digiplan/map/apps.py", line 21, in ready
    from digiplan.map import hooks as digiplan_hooks
  File "/app/digiplan/map/hooks.py", line 10, in <module>
    from digiplan.map import config, datapackage, forms
  File "/app/digiplan/map/forms.py", line 65, in <module>
    class PanelForm(TemplateExtraContentForm):  # noqa: D101
  File "/app/digiplan/map/forms.py", line 91, in PanelForm
    def generate_fields(self, parameters: dict, additional_parameters: dict | None = None) -> dict:
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```
I merged the `dev` which included #59 but this shouldn't be the problem?!

**Comment by nesnoj** [comment]
Or maybe it's simply not finished? :D

**Comment by henhuy** [comment]
This is due to your python version - seems outdated.
I will update python version in dockerfile in separate PR

**Comment by henhuy** [comment]
Fixed it with future annotation

**Comment by nesnoj** [comment]
> Fixed it with future annotation

Great, error vanished.
However, the values do not change on slider adjustments but I guess this is the part you will integrate as soon as we decide for ion/AJAX way?

**Comment by nesnoj** [comment]
Ok, as there's no progress at ion, please implement using ajax.

**Review by nesnoj — COMMENTED** [review]
I rebuild the docker but I'm getting `ModuleNotFoundError: No module named 'template_partials'` here..

File: poetry.lock [hunk 1]
```diff
@@ -1175,6 +1189,24 @@ files = [
 Django = ">=2.2"
 redis = ">=3.0.0"
 
+[[package]]
```
Comment (henhuy) [inline poetry.lock:1192]: Strange? Cause `django-template-partials` is added in poetry.locks...
Can't explain right now

**Review by nesnoj — COMMENTED** [review]
Hey @henhuy, great feature nicely integrated!
I fixed the calculations, values are correct now. I have 2 questions:
- wind_2024 (`s_w_6`): the slider value must be divided by its max value, I did this in bb090f83648d7c9f7e26de52f91ebb14e0da831e. Is it ok this way or should the max value rather be obtained from context ?
- PV roof potential: you used the capacity, however, in the current implementation areas must be used. I changed this in ede4085bf4f52d463ab51a80de3682afccb6b7f4. Unfortunately in the pipeline I had used sqm not sqkm like for the other technologies. Therefore I changed the pipeline to get everything in line. You'll need the latest `potentialarea_pv_roof_area_stats_muns.csv` from the [wolke](https://wolke.rl-institut.de/f/738975) to make it work.

If you're fine with my changes, please go ahead and merge..

PS: BTW, the fct [`calculate_potential_shares`](https://github.com/empowerplan/epp-app/blob/81629c72281607b5b16065068030379491ffe9bc/digiplan/map/calculations.py#L515) seems to be outdated..

**Comment by henhuy** [comment]
Your fixes are working. Reading max wind from energy settings panel is totally fine.
I moved adaption of potential shares calculation to new issue #91 
You can merge now

**Review by nesnoj — APPROVED** [review]

**Pull request merged**